lib/embed/embed_this_panel.rb

```diff
@@ -0,0 +1,58 @@
+module Embed
+  class EmbedThisPanel
+    def initialize(druid:, height:, width:, purl_object_title:, &block)
+      @druid = druid
+      @height = height
+      @width = width
+      @purl_object_title = purl_object_title
+      @panel_content = yield block
+    end
+
+    def to_html
+      <<-HTML.strip_heredoc
+        <div class='sul-embed-panel-container'>
+          <div class='sul-embed-panel sul-embed-embed-this-panel' style='display:none' aria-hidden='true'>
+            <div class='sul-embed-panel-header'>
+              <button class='sul-embed-close' data-sul-embed-toggle='sul-embed-embed-this-panel'>
+                <span aria-hidden='true'>&times;</span>
+                <span class='sul-embed-sr-only'>Close</span>
+              </button>
+              <div class='sul-embed-panel-title'>Embed</div>
+            </div>
+            <div class='sul-embed-panel-body'>
+              <div class='sul-embed-embed-this-form'>
+                <span class='sul-embed-options-label'>Select options:</span>
+                <div class='sul-embed-section sul-embed-embed-title-section'>
+                  <input type='checkbox' id='sul-embed-embed-title' 'data-embed-attr'='hide_title' checked=true />
+                  <label for='sul-embed-embed-title'>
+                    title
+                    <span class='sul-embed-embed-title'> (#{purl_object_title})</span>
+                  </label>
+                </div>
+                #{panel_content}
+                <div class='sul-embed-section'>
+                  <input type='checkbox' id='sul-embed-embed' data-embed-attr='hide_embed' checked=true />
+                  <label for='sul-embed-embed'>embed</label>
+                </div>
+                <div>
+                  <div class='sul-embed-options-label'>
+                    <label for='sul-embed-iframe-code'>Embed code:</label>
+                  </div>
+                  <div class='sul-embed-section'>
+                    <textarea id='sul-embed-iframe-code' data-behavior='iframe-code' rows=4>
+                      <iframe src='#{Settings.embed_iframe_url}?url=#{Settings.purl_url}/#{druid}' height='#{height}px' width='#{width || height}px' frameborder='0' marginwidth='0' marginheight='0' scrolling='no' allowfullscreen></iframe>
```

Comment:
Unfortunately it looks like the heredoc spacing in the `<textarea>` is making its way into the actual form.

<img width="521" alt="textarea" src="https://cloud.githubusercontent.com/assets/96776/14578174/f758cbbe-0335-11e6-953e-a93728497466.png">